### ISSUE TITLE
Compilation fails with Rust 1.72

### ISSUE THREAD

**Issue by direc85** [woboq/qmetaobject-rs#299]
```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> qmetaobject/src/scenegraph.rs:158:32
    |
158 |             raw = cpp!(unsafe [type_id as "quint64", len as "std::size_t"] -> *mut c_void as "QSGNode*" {
    |                                ^^^^^^^
    |
    = note: source type: `TypeId` (128 bits)
    = note: target type: `[u8; 8]` (64 bits)
    = note: this error originates in the macro `__cpp_closure_impl` which comes from the expansion of the macro `cpp` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> qmetaobject/src/scenegraph.rs:163:58
    |
163 |             mask = cpp!(unsafe [raw as "ContainerNode*", type_id as "quint64", len as "std::size_t"] -> u64 as "quint64" {
    |                                                          ^^^^^^^
    |
    = note: source type: `TypeId` (128 bits)
    = note: target type: `[u8; 8]` (64 bits)
    = note: this error originates in the macro `__cpp_closure_impl` which comes from the expansion of the macro `cpp` (in Nightly builds, run with -Z macro-backtrace for more info)
(...)
```

This is caused by Rust 1.72 change [Use 128 bits for TypeId hash](https://github.com/rust-lang/rust/pull/109953/)

**Comment by ogoffart** [comment]
This is strange. I thought commit 947a73b4d87b5000a8e5719ad28efc0eea68bff1  which is included in the version 0.2.9 of the crate would have fixed that already

**Comment by direc85** [comment]
Ah, indeed it does. I'm trying to compile Whisperfish, which is stuck around 0.2.5, so this bug is irrelevant - and so is the merge request.
